GoogleForCreators/web-stories-wp
# Performance: Use dynamic imports

## Context
This PR does some code splitting by dynamically importing modules not required when loading the editor at first.

Note: there are still more modules to import dynamically, such as `draft-js` and some others.

## To-do

- [x] html-to-image
- [x] react-color (needed for the color picker only)
- draft-js (seemingly needed for text elements only) [EDIT: WIll be done in a follow-up issue: google/web-stories-wp/issues/8336]
- [x] colorthief (could be loaded only when starting to use, that's happening in the background anyway)
- [x] react-calendar (used for the calendar only)
- [x] react-moveable (not needed for the initial render)

## Testing Instructions

### QA

This PR can be tested by following these steps:

I
1. Add a shape
2. Now set network to slow 3g
3. Open the Color Picker
4. Verify that while the color picker is loading, a spinner is displayed and once it loads, the color picker is not positioned partially out of the screen.

II
1. Open the Document panel.
2. Set the network to slow 3g
3. Open the calendar.
4. Verify that while the calendar is loading, a spinner is displayed, and calendar doesn't "jump" once it loads.


## Checklist



---

Fixes #8251
